kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java

```diff
@@ -237,6 +237,11 @@ public boolean hasMoreLogs() {
       throw new ClosedOrCancelledException(
           "Method getEngineLog() failed. The " + "connection has been closed.");
     }
+
+    if (launchEngineOpHandle == null) {
```

Comment:
Is this handle that caller call `getExecLog()` without check of `hasMoreLogs()`?

Reply:
I see,

it was invoked in
```
  private void showRemainingLogsIfAny(Object sqlObject) {
    if (sqlObject instanceof KyuubiLoggable) {
      KyuubiLoggable kyuubiLoggable = (KyuubiLoggable) sqlObject;
      List<String> logs = null;
      do {
        try {
          logs = kyuubiLoggable.getExecLog();
```

Reply:
here the method name is `showRemainingLogsIfAny`.
and hasMoreLogs will check whether the launch engine operation completed.

So, it does not check hasMoreLogs.

Reply:
Is there any side effect if we check `hasMoreLogs` before call `showRemainingLogsIfAny`?

Reply:
if we check hasMoreLogs here, the `showRemainingLogsIfAny` will not show remaining logs.